src/main/java/ListUtils.java

```diff
@@ -0,0 +1,244 @@
+                saveToFile();
+                System.out.println(" ____________________________________________________________");
+                System.out.println("quag! you're done with this task :");
+                System.out.println("  " + task);
+                System.out.println(" ____________________________________________________________");
+            }
+        } else {
+            System.out.println("invalid task number! quag");
+        }
+    }
+
+    public static void markAsNotDone(int index) {
+        if (index >= 1 && index <= userInputs.size()) {
+            Task task = userInputs.get(index - 1);
+            if (!task.isDone) {
+                System.out.println(" ____________________________________________________________");
+                System.out.println("quag! you ain't even done w this yet:");
+                System.out.println("  " + task);
+                System.out.println(" ____________________________________________________________");
+            } else {
+                task.markAsNotDone();
+                saveToFile();
+                System.out.println(" ____________________________________________________________");
+                System.out.println("quag! you're NOT done with this task :");
+                System.out.println("  " + task);
+                System.out.println(" ____________________________________________________________");
+            }
+        } else {
+            System.out.println("invalid task number! quag");
+        }
+    }
+
+    public static void deleteTask(int taskIndex) {
+        System.out.println(" ____________________________________________________________");
+        System.out.println("quag! deleted this task :");
+        System.out.println("  " + userInputs.get(taskIndex - 1));
+        userInputs.remove(taskIndex - 1);
+        System.out.println("You have " + userInputs.size() + " quaggin tasks to do! get to work!");
+        System.out.println(" ____________________________________________________________");
+        saveToFile();
+    }
+
+    public static void saveToFile() {
+        try {
+            File dir = new File("./data");
+            //create directory if file does not exist
+            if (!dir.exists()) {
+                dir.mkdirs();
+            }
+            FileWriter writer = new FileWriter(FILE_PATH);
+            for (Task task : userInputs) {
+                if (task instanceof Todo) {
+                    writer.write("T | " + (task.isDone ? "1" : "0") + " | " + task.description + "\n");
+                } else if (task instanceof Deadline) {
+                    writer.write("D | " + (task.isDone ? "1" : "0") + " | " + task.description + " | " + ((Deadline) task).by + "\n");
+                } else if (task instanceof Event) {
+                    writer.write("E | " + (task.isDone ? "1" : "0") + " | " + task.description + " | " + ((Event) task).from + " | " + ((Event) task).to + "\n");
```

Comment:
Maybe it would be a good idea to refactor and create a method for each of these if statements to improve code readability.